src/redux/epics/download-epic.js

```diff
@@ -57,23 +57,29 @@ const beginDownload = (action$, store) => {
 
     return Observable.fromPromise(Iota.findTransactions(nonMetaDataAddresses))
       .map(transactions => {
-        const orderedTransactions = _.sortBy(transactions, t =>
-          nonMetaDataAddresses.indexOf(t.address)
+        const addrToIdx = _.reduce(
+          nonMetaDataAddresses,
+          (acc, addr, idx) => {
+            acc[addr] = idx;
+            return acc;
+          },
+          {}
         );
-        const chunksArrayBuffers = orderedTransactions.map(t => {
+        const orderedTransactions = _.sortBy(
+          transactions,
+          tx => addrToIdx[tx.address]
+        );
+        const chunksArrayBuffers = orderedTransactions.map(tx => {
           return FileProcessor.chunkFromIotaFormat(
-            t.signatureMessageFragment,
+            tx.signatureMessageFragment,
             handle
           );
         });
 
         const completeFileArrayBuffer = FileProcessor.mergeArrayBuffers(
           chunksArrayBuffers
         );
-        console.log(
-          "DOWNLOADED ARRAY BUFFER: ",
-          new Uint8Array(completeFileArrayBuffer)
-        );
+        console.log("DOWNLOADED ARRAY BUFFER");
```

Comment:
we want to print `new Uint8Array(completeFileArrayBuffer)` too so that we can easily verify from the console that the file is the same as the one that we uploaded

Reply:
Printing a huge string is very slow though. Are we sure we want to? It won't give us a realistic idea of our speed.

Reply:
hm... ok fine